sklearn/utils/_plotting.py

```diff
@@ -63,6 +80,198 @@ def _validate_from_predictions_params(
 
         return pos_label, name
 
+    @classmethod
+    def _validate_from_cv_results_params(
+        cls,
+        cv_results,
+        X,
+        y,
+        *,
+        sample_weight,
+        pos_label,
+        name,
+        fold_line_kwargs,
+    ):
+        check_matplotlib_support(f"{cls.__name__}.from_predictions")
+
+        required_keys = {"estimator", "indices"}
+        if not all(key in cv_results for key in required_keys):
+            raise ValueError(
+                "'cv_results' does not contain one of the following required keys: "
+                f"{required_keys}. Set explicitly the parameters return_estimator=True "
+                "and return_indices=True to the function cross_validate."
+            )
+
+        train_size, test_size = (
+            len(cv_results["indices"]["train"][0]),
+            len(cv_results["indices"]["test"][0]),
+        )
+
+        if _num_samples(X) != train_size + test_size:
+            raise ValueError(
+                "'X' does not contain the correct number of samples. "
+                f"Expected {train_size + test_size}, got {_num_samples(X)}."
+            )
+
+        if type_of_target(y) != "binary":
+            raise ValueError(
+                f"The target y is not binary. Got {type_of_target(y)} type of"
+                " target."
+            )
+        check_consistent_length(X, y, sample_weight)
+
+        try:
+            pos_label = _check_pos_label_consistency(pos_label, y)
+        except ValueError as e:
+            # Adapt error message
+            raise ValueError(str(e).replace("y_true", "y"))
+
+        n_curves = len(cv_results["estimator"])
+        if (
+            isinstance(name, list)
+            and len(name) != 1
+            and (isinstance(fold_line_kwargs, Mapping) or fold_line_kwargs is None)
+        ):
+            raise ValueError(MULTICURVE_LABELLING_ERROR.format(n_curves=n_curves))
+        else:
+            if isinstance(name, list) and len(name) not in (1, n_curves):
+                raise ValueError(
+                    f"`name` must be a list of length {n_curves} or a string. "
+                    f"Got list of length: {len(name)}."
+                )
+
+        # Add `default_curve_kwargs` to `fold_line_kwargs`
+        default_curve_kwargs = {"alpha": 0.5, "linestyle": "--"}
+        if fold_line_kwargs is None:
+            fold_line_kwargs_ = default_curve_kwargs
+        elif isinstance(fold_line_kwargs, Mapping):
+            fold_line_kwargs_ = _validate_style_kwargs(
+                default_curve_kwargs, fold_line_kwargs
```

Comment:
With this approach, `from_cv_results` calls `validate_from_cv_results_params` which calls `validate_style_kwargs`. Then it calls `plot` which calls `get_line_kwargs` which again calls `validate_style_kwargs`. There's duplicated validation.

That's why I suggested to have all validation of the curve kwargs and incorporation of defaults happening in plot. `from_predictions` doesn't validate the kwargs and directly passes them to `plot`. It would be cleaner and simpler to have a single centralized place to validate the curve kwargs for all methods.

Reply:
Thinking again, I think it makes sense to have these default kwargs always when a user is plotting multiple curves, so will amend